src/resmom/linux/alps.c

```diff
@@ -3778,22 +3616,17 @@ inventory_loop_on_segments(basil_node_t *node, vnl_t *nv, char *arch,
 			 * All labels are assumed to be the same on
 			 * all segments.
 			 */
-			for (proc = seg->processors; proc; proc = proc->next)
-				totcpus++;
 			for (mem = seg->memory; mem; mem = mem->next)
 				totmem += mem->page_size_kb * mem->page_count;
 
 			cu = seg->computeunits;
 			if (cu) {
-				attr = "resources_available.vps_per_ppu";
-				sprintf(utilBuffer, "%d", cu->proc_per_cu_count);
-				if (vn_addvnr(nv, vname, attr, utilBuffer,
-					0, 0, NULL) == -1)
-					goto bad_vnl;
-
 				for (cu = seg->computeunits; cu; cu = cu->next) {
-					totppus++;
+					totcpus++;
 				}
+			} else {
+				for (proc = seg->processors; proc; proc = proc->next)
+					totcpus++;
```

Comment:
Same is true here, you don't need the if (cu) or the loop on cu.  You just need to loop to count the processors.